wicker/core/datasets.py

```diff
@@ -22,6 +24,154 @@
 FILE_LOCK_TIMEOUT_SECONDS = 300
 
 
+def get_file_size_s3_proced(buckets_keys: List[Tuple[int, int]]) -> int:
+    """Get file size of s3 files, most often column files.
+
+    This works on any list of buckets and keys but is generally only
+    used for column files as those are the majority of what is stored on
+    s3 for Wicker. Wicker also stores parquet files on s3 but those are limited
+    to one file per dataset and one schema file.
+
+    This splits your buckets_keys_list across mutliple processes on your local host
+    where each process is then multi threaded further. This reduces the i/o wait to
+    what is assumed to be the absolute minimum that is possible.
+
+    Args:
+        buckets_keys: (List[Tuple[str, str]]): A list of buckets and keys for which
+        to fetch size in bytes on s3. Tuple index 0 is bucket and index 1 is key of the file.
+
+    Returns:
+        int size of file list in bytes.
+    """
+    # declare the variables neccessary to split
```

Comment:
IMHO I don' t think the extra comments here are necessary as you have done a good job describing the intention of what should happen in the function comments above.

Reply:
Done